paimon-core/src/main/java/org/apache/paimon/catalog/AbstractCatalog.java

```diff
@@ -43,13 +52,17 @@ public abstract class AbstractCatalog implements Catalog {
 
     protected final Map<String, String> tableDefaultOptions;
 
+    @Nullable protected final LineageMeta lineageMeta;
+
     protected AbstractCatalog(FileIO fileIO) {
         this.fileIO = fileIO;
+        this.lineageMeta = null;
         this.tableDefaultOptions = new HashMap<>();
     }
 
-    protected AbstractCatalog(FileIO fileIO, Map<String, String> options) {
+    protected AbstractCatalog(FileIO fileIO, Map<String, String> options, ClassLoader classLoader) {
```

Comment:
Can we just use AbstractCatalog.class.getClassLoader? Simple, and the implementation should be togeter with paimon jar.